### DISCUSSION TITLE
Let's make a feature wishlist for v1.0.0!

### DISCUSSION THREAD

**Discussion by carreter** [TimothyStiles/poly#373] [Ideas]
It'd be nice to get this package into a truly releasable (and maybe even publishable? 👀) state. See the [poly development roadmap](https://github.com/users/TimothyStiles/projects/1/views/1?sliceBy%5Bvalue%5D=v1.0) for what we've proposed so far.

We'll be talking about this at our upcoming developer meetups, but I wanted to get the ball rolling!

**Comment by carreter** [comment]
I've already started reaching out to faculty at MIT to ask for their thoughts on this (and to see if they will mentor me). Will relay any info I get here.

**Comment by TimothyStiles** [comment]
I want to make poly more accessible to more ecosystems. @Koeng101 mentioned using wasm and protobuf definitions to generate multi-language bindings. I'm also wondering what kind of codegen tools are out there for generating APIs and CLIs from exposed function signatures. If we can automate those I'd be down for shipping them.

**Reply by Koeng101** [reply]
gRPC is fantastic for generating APIs from function signatures - it is kinda what is built to do. For most codegen though you have to add a little more in to make it idiomatic, but it's usually not difficult. 

What we could do there is use protobuf definitions + wasm to generate boilerplate for every language, and then just layer a bit of idiomatic library code on top.

**Reply by Koeng101** [reply]
If I have some time later I might throw together a little demo of what this would look like

**Comment by Koeng101** [comment]
A very biased list of what I would like to see:

- Parsers standardized
- sam parser merged
- minimap2/samtools integration with wasm
- mutation calling on pileup files
- gibson assembly
- improved cloning / ligation simulation
- protobuf definitions for all parsers
- Full package wasm compatibility + simple integration test

Lot of things, but I think overall would make a very compelling bioinformatics package + would really close the loop on the whole clone pipeline in a way we could port to python and such.

**Comment by carreter** [comment]
I really want to have the basics of structural analysis tools in poly before v1.0. Not sure what the would look like, but #297 is definitely a start!
